substrate/frame/examples/basic/src/lib.rs

```diff
@@ -177,267 +188,199 @@ pub mod pallet {
+	#[pallet::getter(fn dummy_value_query)]
+	pub(super) type DummyValueQuery<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	/// An example storage map that has enumerable entries. In our case this is a mapping of
+	/// AccountIds to Balances.
+	///
+	/// We don't actually use this anywhere.
+	#[pallet::storage]
+	#[pallet::getter(fn accounts_to_balances_map)]
+	pub(super) type AccountsToBalancesMap<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, T::Balance>;
+
+	/// The genesis configuration type.
+	#[pallet::genesis_config]
+	#[derive(frame_support::DefaultNoBound)]
+	pub struct GenesisConfig<T: Config> {
+		pub dummy: T::Balance,
+		pub dummy_value_query: u32,
+		pub accounts_to_balances_map: Vec<(T::AccountId, T::Balance)>,
+	}
+
+	/// The genesis build for this pallet.
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {
+			Dummy::<T>::put(&self.dummy);
+			DummyValueQuery::<T>::put(&self.dummy_value_query);
+			for (a, b) in &self.accounts_to_balances_map {
+				AccountsToBalancesMap::<T>::insert(a, b);
+			}
 		}
 	}
 
-	// The call declaration. This states the entry points that we handle. The
+	/// Events are a simple means of reporting specific conditions and
+	/// circumstances that have happened that users, Dapps and/or chain explorers would find
+	/// interesting and otherwise difficult to detect.
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// The Dummy value has been accumulated.
+		AccumulateDummy {
+			/// The new value of Dummy.
+			balance: BalanceOf<T>,
+		},
+		/// The Dummy value has been set.
+		SetDummy {
+			/// The new value of Dummy.
+			balance: BalanceOf<T>,
+		},
+	}
+
+	// The call declaration block of our pallet. This states the entry points that we handle. The
 	// macro takes care of the marshalling of arguments and dispatch.
 	//
-	// Anyone can have these functions execute by signing and submitting
-	// an extrinsic. Ensure that calls into each of these execute in a time, memory and
-	// using storage space proportional to any costs paid for by the caller or otherwise the
-	// difficulty of forcing the call to happen.
-	//
-	// Generally you'll want to split these into three groups:
-	// - Public calls that are signed by an external account.
-	// - Root calls that are allowed to be made only by the governance system.
-	// - Unsigned calls that can be of two kinds:
-	//   * "Inherent extrinsics" that are opinions generally held by the block authors that build
-	//     child blocks.
-	//   * Unsigned Transactions that are of intrinsic recognizable utility to the network, and are
-	//     validated by the runtime.
+	// Each call must define a `#[pallet::weight(..)]` attribute to convey a set of
+	// static information about its dispatch. The FRAME System and FRAME Executive pallets then use
+	// this information to properly execute transaction, whilst keeping the total load of
```

Comment:
The main  point of the weight system is to know the **pre-dispatch** weight of stuff. where did 

> whilst keeping the total load of the chain in a moderate rate

Come from? We could also keep the load moderate if we do metering. What would be the difference? 

I am happy to simplify things, but the simplification must be rooted at reality.